pkg/orchestrator/scheduler/batch_job_test.go

```diff
@@ -60,11 +61,71 @@ func TestBatchSchedulerTestSuite(t *testing.T) {
 	suite.Run(t, new(BatchJobSchedulerTestSuite))
 }
 
+func (s *BatchJobSchedulerTestSuite) TestProcess_ShouldDelayAfterRejections() {
+	ctx := context.Background()
+	job, executions, evaluation := mockJob()
+	job.Count = 1
+	// Just one execution, rejected very recently
+	executions = []models.Execution{
+		executions[0],
+	}
+	executions[0].ComputeState = models.NewExecutionState(models.ExecutionStateAskForBidRejected)
+	executions[0].CreateTime = time.Now().UnixNano()
+	s.jobStore.EXPECT().GetJob(gomock.Any(), job.ID).Return(*job, nil)
+	s.jobStore.EXPECT().GetJobHistory(gomock.Any(), job.ID, gomock.Any()).Return([]models.JobHistory{}, nil)
+	s.jobStore.EXPECT().GetExecutions(gomock.Any(), jobstore.GetExecutionsOptions{JobID: job.ID}).Return(executions, nil)
+	s.jobStore.EXPECT().RecordJobDeferral(gomock.Any(), job.ID, gomock.Any(), "Deferring rescheduling for 10s")
+
+	// No nodes are suitable, as we're within the retry delay of the last failure
+	nodeInfos := []models.NodeInfo{}
+
+	s.mockNodeSelection(job, nodeInfos, job.Count, 10*time.Second)
+
+	matcher := NewPlanMatcher(s.T(), PlanMatcherParams{
+		Evaluation:         evaluation,
+		NewEvaluationDelay: time.Duration(10 * time.Second),
```

Comment:
better to create an evaluation with expected type and delay, and pass that to the matcher to verify that a new matching evaluation has been enqueued

Reply:
Done